### PR TITLE
Add docs FAQ entry for special html characters

### LINKED ISSUE #1098: <= in Question HTML Breaks Rendering
Something in PL seems to be interpreting a `<=` in `question.html` as part of a tag. Unfortunately this breaks rendering of many code snippets that include a `<=`.

Example `question.html`:
````html
<pl_question_panel>                                                                                                                                                                  
  <pre>                                                                                                                                                                              
    <code>                                                                                                                                                                           
for (int i = 0; i <= 10; i++);                                                                                                                                                       
    </code>                                                                                                                                                                          
  </pre>                                                                                                                                                                             
  <p>Is the question above broken?</p>                                                                                                                                               
    <pl_multiple_choice weight="1" answers_name="Sp18_Broken_0" number_answers="2">                                                                                                  
        <pl_answer correct="true">Yes</pl_answer>                                                                                                                                    
        <pl_answer correct="false">No</pl_answer>                                                                                                                                    
    </pl_multiple_choice>                                                                                                                                                            
</pl_question_panel> 
````
Result:

image

If you replace the `<=` with a `<`, all is well:

````html
<pl_question_panel>                                                                                                                                                                  
  <pre>                                                                                                                                                                              
    <code>                                                                                                                                                                           
for (int i = 0; i < 10; i++);                                                                                                                                                       
    </code>                                                                                                                                                                          
  </pre>                                                                                                                                                                             
  <p>Is the question above broken?</p>                                                                                                                                               
    <pl_multiple_choice weight="1" answers_name="Sp18_Broken_0" number_answers="2">                                                                                                  
        <pl_answer correct="true">Yes</pl_answer>                                                                                                                                    
        <pl_answer correct="false">No</pl_answer>                                                                                                                                    
    </pl_multiple_choice>                                                                                                                                                            
</pl_question_panel> 
````
image

[Here's a link to an example on the main PL site](https://prairielearn.engr.illinois.edu/pl/course_instance/8235/instructor/question/1541331/). Note that it's markup is a bit different since we preprocess things through HLJS, but I've verified that the minimal example above still triggers the problem.

I suspect that something is interpreting a `<=` as the start of a template tag. It seems like at minimum this could be disabled in `code` or `pre` blocks.

### PR DESCRIPTION
**Pull request by nicknytko**
Special characters like `<`, `>`, `&` can break rendering on the page especially when used in inline code segments.  This is common enough that we should probably write something on the docs -- this adds an entry in the FAQ and a small section in the elements page for `pl-code`.

Resolves #1098

### CODE REVIEW

**Comment by echuber2** [comment]
Do you want to add a note that this doesn't apply to embedded markdown? E.g. the recommendations are to either use pl-code with manual escaping, or use it with external (raw) code files, or use the embedded markdown (without manual escaping).

**Comment by echuber2** [comment]
Now that I say that, I forget whether this affects embedded markdown too or not. I know it gets parsed out and manually escaped, but maybe that only solves problems with things like ampersands, not angle brackets that look like closing tags.

**Comment by nicknytko** [comment]
> Do you want to add a note that this doesn't apply to embedded markdown? E.g. the recommendations are to either use pl-code with manual escaping, or use it with external (raw) code files, or use the embedded markdown (without manual escaping).

Good point, I think markdown will correctly escape ampersands and angle brackets as well if they're in a code block.  I suspect it will also escape characters outside of a code block, but I'll test this to make sure.

File: docs/elements.md [hunk 1]
```diff
@@ -773,6 +773,10 @@ Attribute | Type | Default | Description
 
 The `pl-code` element uses the _Pygments_ library for syntax highlighting, a full list of supported languages can be found [here](https://pygments.org/languages/).
 
+##### Common Pitfalls
+
+The HTML specification disallows inserting special characters onto the page (i.e. `<`, `>`, `&`), and using these characters with inline code may break rendering.  To fix this, either escape the characters (`&lt;`, `&gt;`, `&amp;`, more [here](https://www.freeformatter.com/html-entities.html)), or load code snippets from external files.
```
Comment (coatless) [inline docs/elements.md:778]:
```suggestion
The HTML specification disallows inserting special characters onto the page (i.e. `<`, `>`, `&`), and using these characters with inline code may break rendering.  To fix this, either escape the characters (`&lt;`, `&gt;`, `&amp;`, more [here](https://www.freeformatter.com/html-entities.html)), or load code snippets from external files into `pl-code` with `source-file-name` attribute.
```

File: docs/faq.md [hunk 2]
```diff
@@ -282,3 +282,7 @@ To address this, there are a variety of different ways. In particular, we have:
 - Restart your computer.
 - Stop the process in terminal with <kbd>CNTRL</kbd> + <kbd>C</kbd> and, then, 
   close the terminal application.   
+
+## Why do special characters like (<=) break my question display?
+
+The HTML specification disallows inserting special characters onto the page (i.e. `<`, `>`, `&`), and using these characters in your question, for example with inline code, may break rendering.  To fix this, either escape the characters (`&lt;`, `&gt;`, `&amp;`, more [here](https://www.freeformatter.com/html-entities.html)), or load code snippets from external files.
```
Comment (coatless) [inline docs/faq.md:288]:
```suggestion
The HTML specification disallows inserting special characters onto the page (i.e. `<`, `>`, `&`), and using these characters in your question, for example with inline code, may break rendering.  To fix this, either escape the characters (`&lt;`, `&gt;`, `&amp;`, more [here](https://www.freeformatter.com/html-entities.html)), or load code snippets from external files into `pl-code` with `source-file-name` attribute. For more information, see the [`pl-code` element documentation](elements.md#pl-code-element)
```

**Pull request merged**